waspc/src/Wasp/Generator/DbGenerator/Operations.hs

```diff
@@ -116,3 +133,19 @@ doesSchemaMatchDb genProjectRootDirAbs = do
     ExitSuccess -> return $ Just True
     ExitFailure 2 -> return $ Just False
     ExitFailure _ -> return Nothing
+
+-- | Checks `prisma migrate status` exit code to determine if migrations dir
+-- matches the DB. Returns Nothing on error as we do not know the current state.
+-- Returns Just True if all migrations are applied. Due to the fact the command
+-- returns an error on connection or unapplied migrations, Just False is never returned.
+-- It is recommended to call this after some check that confirms DB connectivity, like `doesSchemaMatchDb`.
+areAllMigrationsAppliedToDb :: Path' Abs (Dir ProjectRootDir) -> IO (Maybe Bool)
+areAllMigrationsAppliedToDb genProjectRootDirAbs = do
+  chan <- newChan
+  (_, dbExitCode) <-
+    concurrently
+      (readJobMessagesAndPrintThemPrefixed chan)
+      (DbJobs.migrateStatus genProjectRootDirAbs chan)
+  case dbExitCode of
+    ExitSuccess -> return $ Just True
+    ExitFailure _ -> return Nothing
```

Comment:
Yeah this is tricky. I sorta liked it _not_ being just two states to explicitly capture there is some ambiguity in how the command returns the status. And it is similar to another command that does indicate all 3 states. My hope in the future is prisma takes my feedback and makes all their commands able to have these 3 states, so it was: a) documentation of the ambiguity, b) consistency with the one above, and c) hopefully planning ahead for the fix. For those reasons, I think it may make sense to leave as is, sound ok?

Reply:
Sounds good!